op-challenger/fault/monitor.go

```diff
@@ -147,13 +143,23 @@ func (m *gameMonitor) fetchOrCreateGamePlayer(gameData FaultDisputeGame) (gamePl
 func (m *gameMonitor) MonitorGames(ctx context.Context) error {
 	m.logger.Info("Monitoring fault dispute games")
 
+	blockNum := uint64(0)
 	for {
-		err := m.progressGames(ctx)
-		if err != nil {
-			m.logger.Error("Failed to progress games", "err", err)
-		}
-		if err := m.clock.SleepCtx(ctx, 300*time.Millisecond); err != nil {
-			return err
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			nextBlockNum, err := m.fetchBlockNumber(ctx)
+			if err != nil {
+				m.logger.Error("Failed to load current block number", "err", err)
+				continue
+			}
+			if nextBlockNum > blockNum {
+				blockNum = nextBlockNum
+				if err := m.progressGames(ctx, nextBlockNum); err != nil {
+					m.logger.Error("Failed to progress games", "err", err)
+				}
+			}
```

Comment:
This is now a tight loop constantly polling the L1 head which isn't great. We should put the sleep back in (and probably make it sleep for say 1 second rather than 300ms).

With the sleep back in we can probably merge this as-is, but I'd probably keep the ticket open as we should switch to using a subscription to track the head (can reuse the same code as op-node uses to track L1 head) to avoid constantly polling. We've avoided subscriptions for log events because of the risk of missing one, but that's not an issue for L1 head because we'll just get the next head update and automatically catch everything up again.

Reply:
Sounds good to me. I'm happy to follow up with a subscription based pr.